README.md

```diff
@@ -1 +1,401 @@
+Method: PUT
+Request Body:
+```json
+{
+    "providerId": "",
+    "role": "client"
+}
+```
+
+6. Create Client Profile
+Description: Creates a new client profile.
+Endpoint: /clients/
+Method: POST
+Request Body:
+```json
+{
+  "address": "",
+  "contactNumber": ""
+}
+```
+
+7. Get Client Profile
+Description: Retrieves the profile of the authenticated client.
+Endpoint: /clients/me
+Method: GET
+Request Body:
+```json
+{
+  
+}
+```
+
+8. Update Client Profile
+Description: Updates the profile of the authenticated client.
+Endpoint: /clients/me
+Method: PUT
+Request Body:
+```json
+{
+  "address": "",
+  "contactNumber":""
+}
+```
+
+
+9. Retrieve Chefs by a client
+Description: Retrieves a list of all chefs by a client.
+Endpoint: clients/chefs
+Method: GET
+Request Body:
+```json
+{
+ 
+}
+```
+10. Add Dish to Client's Favorite Dishes
+Description: Adds a dish to the list of favorite dishes for the authenticated client.
+Endpoint: /clients/me/favorite-dishes
+Method: POST
+Request Body:
+```json
+{
+  "dishId": ""
+}
+```
+
+11. Retrieve Client's Favorite Dishes
+Description: Retrieves a list of dishes favorited by the authenticated client.
+Endpoint: /clients/me/favorite-dishes
+Method: GET
+Request Body:
+```json
+{
+  
+}
+```
+
+12. Delete Dish from Client's Favorite Dishes
+Description: Deletes a dish from the list of favorite dishes for the authenticated client.
+Endpoint: /clients/me/favorite-dishes/:id
+Method: DELETE
+Parameters:
+id (required): The ID of the dish to remove from favorites.
+Request Params:
+```json
+{
+  "dishId": ""
+}
+```
+
+13. Retrieve Client's Order History
+Description: Retrieves the order history of the authenticated client.
+Endpoint: /clients/me/order-history
+Method: GET
+Request Body:
+```json
+{
+  
+}
+```
+14. Dish Ratings by Client
+Description: Allows a client to submit ratings for a dish.
+Endpoint: /clients/rating-dish/:id
+Method: POST
+Parameters:
+id (required): The ID of the dish to rate.
+Request Body:
+```json
+{
+  "rating": ""
+}
+```
+
+15. Get Chef Dishes by Client
+Description: Retrieves a list of dishes offered by a specific chef, based on a client's preferences.
+Endpoint: /clients/chefs/dishes/:id
+Method: GET
+Parameters:
+id (required): The ID of the chef to retrieve dishes from.
+
+16. Get Chef By Id
+Description: Retrieves information about a specific chef by their ID.
+Endpoint: /clients/chefs/:id
+Method: GET
+Parameters:
+id (required): The ID of the chef to retrieve.
+
+17. Create Chef Profile
+Description: Allows a user to create a chef profile to showcase their culinary offerings.
+Endpoint: /chef/
+Method: POST
+Request Body:
+```json
+{
+  
+    "restaurant": "",
+    "location": "",
+    "openingHours": "",
+    "closingHours": "",
+    "contactNumber": "",
+    "description": ""
+
+}
+```
+
+18. Edit Chef Profile
+Description: Allows the authenticated chef to edit their profile information.
+Endpoint: /chef/me
+Method: PUT
+Request Body:
+```json
+{
+  "restaurant": ""
```

Comment:
i think we need to add 
 "location": "",
    "openingHours": "",
    "closingHours": "",
    "contactNumber": "",
    "description": ""